src/components/Quote.js

```diff
@@ -0,0 +1,46 @@
+import { useEffect, useState } from 'react';
+import './Quote.css';
+
+function Quote() {
+  const [quote, setQuote] = useState(null);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    fetch('https://api.api-ninjas.com/v1/quotes?category=knowledge', {
+      headers:
+        { 'X-Api-Key': 'A5gXbih2P+SXXKb1RLdvfw==7imgYlRKUhRp6KRD' },
+    })
+      .then((resp) => resp.json())
+      .then((apiData) => {
+        setLoading(false);
+        const body = `${apiData[0].quote} - ${apiData[0].author}`;
+        setQuote(body);
+      })
+      .catch((error) => {
+        setLoading(false);
+        setError(error);
+      });
+  }, []);
```

Comment:
[OPTIONAL] This is optional but recommended so please don't skip this requirement. Well done :clap: you did a very nice job :+1: but I recommended using `async` and `await` because the purpose of `async / await` is to simplify the syntax necessary to consume promise-based APIs. The behaviour of `async / await` is similar to combining generators and promises. `Async` functions always return a promise. Also, create a function and call that function in the `useEffect` then your code looks clean :+1: check below :point_down:  my suggestion :slightly_smiling_face: 

```suggestion
useEffect(() => {
  const fetchData = async () => {
    try {
      const response = await fetch('https://api.api-ninjas.com/v1/quotes?category=knowledge', {
        headers: {
          'X-Api-Key': 'A5gXbih2P+SXXKb1RLdvfw==7imgYlRKUhRp6KRD'
        },
      });
      const apiData = await response.json();
      const body = `${apiData[0].quote} - ${apiData[0].author}`;
      setQuote(body);
    } catch (error) {
      setError(error);
    } finally {
      setLoading(false);
    }
  };

  fetchData();
}, []);
```

In the refactored code, I've converted the `useEffect` callback function into an `async` function called `fetchData`. This allows us to use the `async/await` syntax for cleaner and more readable `asynchronous` code. Inside `fetchData`, I've replaced the `.then` syntax with `await` for fetching the data and parsing the JSON response.

I've also moved the `setLoading(false)` call to the `finally` block to ensure it gets executed whether the API call succeeds or fails. Additionally, I've removed the unnecessary braces around the single statement inside the `catch` block for simplicity.

Finally, I've called the `fetchData` function immediately inside the `useEffect` hook to trigger the API request when the component mounts. The empty dependency `array []` ensures that this effect only runs once.

Reply:
Hello, @aamirkhan2478,

I sincerely appreciate your valuable time and effort in reviewing my pull request. The explanation you have provided is truly insightful, and I am grateful for your guidance. I assure you that I will diligently consider your advice whenever I need to fetch data in the future.

Thank you once again for your assistance.

Reply:
My pleasure bro :slightly_smiling_face: